tools/approval-service/internal/service/githublabels.go

```diff
@@ -0,0 +1,166 @@
+/*
+ *  Copyright 2025 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package service
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/gravitational/shared-workflows/tools/approval-service/internal/eventsources/githubevents"
+	"github.com/gravitational/teleport/api/types"
+)
+
+// githubWorkflowLabels holds information about a GitHub workflow run that is waiting for approval.
+type githubWorkflowLabels struct {
+	// Org is the GitHub organization name.
+	Org string
+	// Repo is the GitHub repository name.
+	Repo string
+	// Env is the environment name for the deployment.
+	Env string
+	// WorkflowRunID is the ID of the workflow run that is waiting for approval.
+	WorkflowRunID int64
+}
+
+// MissingLabelError is returned when a required label is missing from the access request.
+// This error is used to indicate that the access request does not contain the necessary labels to retrieve or store GitHub workflow information.
+// In some cases, this is not an error, but rather a signal that the access request needs to be updated with the required labels.
+type MissingLabelError struct {
+	// RequestID is the ID of the access request that is missing the label.
+	RequestID string
+	// LabelsKeys is the list of label keys that are required but missing from the access request.
+	LabelKeys []string
+}
+
+// Error implements the error interface for MissingLabelError.
+func (m *MissingLabelError) Error() string {
+	return fmt.Sprintf("access request %q is missing required labels: %s", m.RequestID, strings.Join(m.LabelKeys, ", "))
+}
+
+const (
+	workflowRunLabel  = "workflow_run_id"
+	organizationLabel = "organization"
+	repositoryLabel   = "repository"
+	environmentLabel  = "environment"
+)
+
+func setWorkflowLabels(req types.AccessRequest, info githubWorkflowLabels) error {
```

Comment:
Add godocs to set/get labels methods to explain why we're setting labels on the access request.